src/java.base/share/classes/java/lang/StackTraceElement.java

```diff
@@ -355,27 +354,46 @@ public boolean isNativeMethod() {
      * @revised 9
      * @see    Throwable#printStackTrace()
      */
+    @Override
     public String toString() {
-        String s = "";
-        if (!dropClassLoaderName() && classLoaderName != null &&
-                !classLoaderName.isEmpty()) {
-            s += classLoaderName + "/";
+        int estimatedLength = Objects.requireNonNullElse(classLoaderName, "").length() + 1
+                + Objects.requireNonNullElse(moduleName, "").length() + 1
+                + Objects.requireNonNullElse(moduleVersion, "").length() + 1
+                + declaringClass.length() + 1
+                + methodName.length() + 1
+                + Math.max("Unknown Source".length(), Objects.requireNonNullElse(fileName, "").length()) + 1
```

Comment:
I suggest to define a static variable = "Unknown Source" be used here and line 387.   Same for "Native Method" which can also be accounted for.

Reply:
pulled out to private static final fields